examples/TorchDynamo/pippy_dynamo.py

```diff
@@ -0,0 +1,236 @@
+    nstages = len(list(gm.children()))
+    if expected_stages > 0:
+        assert (
+            nstages == expected_stages
+        ), f"Model is split into {nstages} instead of {expected_stages} stages"
+
+    print(f"\n======= GraphModule after Auto-split =======")
+    print(gm)
+
+    for i, submod in enumerate(gm.children()):
+        print(f"\n======= Child module {i} =======")
+        print(submod)
+
+
+# For storing reference output from inside compiler
+ref_out = ()
+
+
+def run_master(_, args):
+    # PiPPy parameters
+    MULTI_USE_PARAM_CONFIG = (
+        MultiUseParameterConfig.REPLICATE
+        if args.replicate
+        else MultiUseParameterConfig.TRANSMIT
+    )
+    print(f"REPLICATE config: {args.replicate} -> {MULTI_USE_PARAM_CONFIG}")
+    print("Using schedule:", args.schedule)
+
+    # Model parameters
+    d_hid = 512
+    bs = 503
+
+    # Chunking parameters
+    args_chunk_spec = (TensorChunkSpec(0),)
+    kwargs_chunk_spec: Dict = {}
+    output_chunk_spec = (TensorChunkSpec(0),)
+    chunks = 1
+
+    # Ask Dynamo to let PiPPy annotation stay in graph
+    torchdynamo.allow_in_graph(pipe_split)
+
+
+    # Define a compiler backend made by PiPPy for use by Dynamo
+    # The backend comprising:
+    # - PiPPy's graph split, and
+    # - PiPPy's driver creation
+    # The driver is return as a compiled runtime callable, which will be used in the actual data execution
+    def my_pippy_compiler(gm: torch.fx.GraphModule, example_inputs, **kwargs):
+        print("\n============= my_pippy_compiler() called with FX graph =============")
+        gm.graph.print_tabular()
+
+        # Run example input
+        input = example_inputs[0]
+        # Make global for equivalence comparison later
+        global ref_out
+        ref_out = gm(input)
+
+        # PiPPy Pipe creation
+        # Here we split the graph
+        pipe = Pipe.from_tracing(gm, MULTI_USE_PARAM_CONFIG)
+
+        inspect_split_module(pipe, 4)
+
+        # Create PipelineDriver
+        pipe_driver: PipelineDriverBase = schedules[args.schedule](
+            pipe,
+            chunks,
+            args_chunk_spec,
+            kwargs_chunk_spec,
+            output_chunk_spec,
+            args.world_size,
+            _debug_mask_minibatches=True,
+            _record_mem_dumps=bool(args.record_mem_dumps),
+            checkpoint=bool(args.checkpoint),
+        )
+
+        # Return a runtime Callable
+        # This PipelineDriver is a distributed runtime
+        return pipe_driver
+
+
+    class ExampleCode(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.mm_param = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+            self.mm_param2 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+            self.lin = torch.nn.Linear(d_hid, d_hid)
+            self.register_buffer("buffer", torch.randn(bs + 100, d_hid))
+
+        # Decorate with Dynamo
+        @torchdynamo.optimize(my_pippy_compiler)
```

Comment:
I would suggest to remove this

and later write 

opt_ec = torchdynamo.optimize(my_pippy_compiler)(ec)
and then replace later uses of `ec` with `opt_ec`

Reply:
Thanks. Changed :)